jobbergate-api/jobbergate_api/tests/apps/job_submissions/test_routers.py

```diff
@@ -58,19 +59,32 @@ async def test_create_job_submission__with_client_id_in_token(
     create_data.pop("status", None)
     create_data.pop("client_id", None)
 
-    with time_frame() as window:
-        response = await client.post("/jobbergate/job-submissions/", json=create_data)
+    with mock.patch(
+        "jobbergate_api.apps.job_submissions.routers.get_job_properties_from_job_script"
+    ) as mocked:
+        mocked.return_value = JobProperties.parse_obj(create_data["execution_parameters"])
+        with time_frame() as window:
+            response = await client.post("/jobbergate/job-submissions/", json=create_data)
+        mocked.assert_called_once_with(
+            inserted_job_script_id,
+            **create_data["execution_parameters"],
+        )
```

Comment:
still that conversation about mocking this. Is this really needed? If so, why? Because you're fixing how the execution parameters are set, it disables any other developer (even yourself) of doing the same thing but with a different approach.

Reply:
`get_job_properties_from_job_script` is mocked because internally it obtains the job script file from s3 in order to extract the SBATCH parameters. Here the choice was to mock just the high-level function since `get_job_properties_from_job_script` has its own tests mocking low-level functions on another test file.

Reply:
I see. Instead of mocking, have you thought about instantiating a minio server during the tests? This is something I do [here](https://github.com/omnivector-solutions/armada-api/blob/main/docker-compose.yaml#L30-L86) to test against real server. I know such an implementation is quite far from the scope of the PR but it doesn't seem ideal to me the way it is. What do you think? I could help to implement it in parallel why your work on this PR

Reply:
It is a good point, but I agree in parts. Unit tests should test individual components and run as fast as possible to provide us with quick feedback, making the necessary setup and request to external services does not help with that, so I'm in favor of mocking them.

Integration tests, on the other hand, are a good place for that and I agree Jobbergate could make good use of them. Your example is a good use of the concept, and I have another one I've worked with [here](https://github.com/fschuch/file_storehouse/blob/main/tests/test_user_story.py). I'm happy to have a new task covering this topic.

Reply:
> making the necessary setup and request to external services do not help with that,

Quite the opposite. Testing against a real database will give you the exact behaviour you'd have in real deployments.

About the speed: jobbergate tests are slower than the tests I sent you... I still can't see how using a database for testing cannot be beneficial over than the way it currently is.

As I said, this implementation is far from the scope of the task itself but definitely it can be improved. I'm not going to stop your development but I'd much appreciate if we focus on improving that in the near future, and the reason is simple: the way it is doesn't test real scenarios. It only tests if you called a function or not but the implementation is possible in countless ways.

Reply:
I appreciate the comments, I have no doubt we can converge on a solution in the near future, but at the same time, I think we agree it is out of the scope of this PR.

Reply:
do you mind writting a task for that?